frontend/src/store/playbackManager.ts

```diff
@@ -1124,6 +1197,38 @@ class PlaybackManagerStore {
       }
     );
 
+    watch(
+      () => this.currentAudioStreamIndex,
+      (newVal, oldVal) => {
+        if (oldVal !== undefined && newVal !== undefined && oldVal !== newVal) {
+          this.fetchCurrentMediaSource();
+        }
+      }
+    );
+
+    watch(
+      () => ({
+        currentSubtitleStreamIndex: this.currentSubtitleStreamIndex,
+        currentSubtitleTrack: this.currentSubtitleTrack
+      }),
```

Comment:
Can't we merge those watchers together? On both we need to do the same, fetch a new media source.

Reply:
I preferred not to as that'd not be an easy to understand if clause haha. But I don't mind either, as you wish.